### PR TITLE
WIP: Vacuum: Provide human readable fan speeds

### PR DESCRIPTION
**Pull request by syssi**

### CODE REVIEW

File: miio/vacuum.py [hunk 1]
```diff
@@ -306,19 +334,34 @@ def disable_dnd(self):
         return self.send("close_dnd_timer", [""])
 
     @command(
-        click.argument("speed", type=int),
+        click.argument("speed", type=str),
     )
-    def set_fan_speed(self, speed: int):
+    def set_fan_speed(self, speed: str):
```
Comment (syssi) [inline miio/vacuum.py]: I would prefer to pass a Enum here. @rytilahti any suggestions?

**Comment by poma** [comment]
The new vacuum still supports setting speed in percentage (0-100) like in the old one, and it can still return percentage values that will not fit any of predefined values. So methods that work with raw values are still needed.

**Comment by syssi** [comment]
Thanks for the hint! I will update my proposal soon!

**Comment by tribut** [comment]
Not sure if this is the correct place to report it, but given that #523 is closed -
for the Xiaowa E2, the official app sets the fan speeds as following:

~~~
Silent 50
Standard 68
Strong 79
Max 100
~~~

**Comment by rytilahti** [comment]
@syssi should we close this now that #643 is merged in?

**Comment by rytilahti** [comment]
This is obsolete now.

**Pull request closed without merging**